### PR TITLE
Svg added subject preference&working at home

### PR DESCRIPTION
**Pull request by henneboy**
Requested by @mads256h & @Mati-AAU 
Also cleaned up some code

### CODE REVIEW

**Comment by henneboy** [comment]
commit
Did this

File: src/visualizer/svg.js [hunk 1]
```diff
@@ -15,26 +17,49 @@ const {
...
-        }
-        const svgHeight = 2  * svgLineSpace * learningStyles.length;
-        svg.setAttribute("width", svgWidth);
-        svg.setAttribute("height", svgHeight);
-        svg.appendChild(createRect(0, 0, svgWidth, svgHeight)); // create a lightgrey rect that fills the whole svg to show it in browser.
+        const svgDiv = document.createElement("div");
+        svgDiv.setAttribute("class", "svg");
+        svgDiv.setAttribute("style", "width: " + svgWidth + "px;");
+        // Create LearningStyles:
+        const LSTitle = document.createElement("p");
```
Comment (mads256h) [inline src/visualizer/svg.js:33]: Ikk start med stort ;)

**Review by mads256h — CHANGES_REQUESTED** [review]

File: src/visualizer/svg.js (hunk 1 above)
Comment (henneboy) [reply src/visualizer/svg.js]: Omdøbt til learningStyleTitle 👍

File: src/visualizer/statisticsMath.js [hunk 2]
```diff
@@ -73,4 +75,24 @@
     function getLSValuesOfGroup(group, learnStyleName){
         return group.students.map((s) => s.criteria.learningStyles[learnStyleName]);
     }
+
+    /**
+     * @summary Creates and returns a new array with the subject scores
+     * of the subject with the subject name provided in the subjectName argument
+     * @param {object} group the group from which to get the data
+     * @param {string} subjectName the name of the subject
+     * @returns {object} returns an object with the students of the groups values in the given subjects, the key is the students idx in the groups.students object
```
Comment (mads256h) [inline src/visualizer/statisticsMath.js:84]: Det er et number array der bliver returned

File: src/visualizer/statisticsMath.js [hunk 3]
```diff
@@ -73,4 +75,24 @@
...
+     */
+    function getSubjectValuesOfGroup(group, subjectName){
+        return group.students.map((s) => s.criteria.subjectPreference.subjects[subjectName].score);
+    }
+
+    /**
+     * @summary Creates and returns a new array with the workingAtHome scores of the students in the group
+     * @param {object} group the group from which to get the data
+     * @returns {object} returns an object with the students of the groups values in workingAtHome, the key is the students idx in the groups
```
Comment (mads256h) [inline src/visualizer/statisticsMath.js:93]: number array

File: src/visualizer/svg.js [hunk 4]
```diff
@@ -15,26 +17,49 @@ const {
     const svgLineSpace = svgWidth * 0.04; // the amount of space between each line, by the width of the svg
     const svgLineWidth = svgWidth * 0.01;
     const svgTextSize = svgLineSpace * 0.5;
-    const RANGEWIDTH = 22;
+    const LSRANGEWIDTH = 22; // Learningstyle rangewith, from -11 to 11 = 22
+    const WAHRANGEWIDTH = 2; // Working At Home rangewith, from -1 to 1 = 2
 
     /**
      * @summary Creates and appends a div element with the svg of the learningstyles
      * @param {object} group the group array with students
```
Comment (mads256h) [inline src/visualizer/svg.js:25]: Det er vel bare en group og ikke en group array

File: src/visualizer/svg.js [hunk 5]
```diff
@@ -15,26 +17,49 @@ const {
...
+        const workingAtHomeTitle = document.createElement("p");
+        workingAtHomeTitle.innerText = "Working at home";
+        svgDiv.appendChild(workingAtHomeTitle);
+        svgDiv.appendChild(createWorkingAtHome(group));
+        return svgDiv;
+    }
+
+    /**
+     * @summary Creates and appends a div element with the svg of the learningstyles
+     * @param {object} group the group array with students
```
Comment (mads256h) [inline src/visualizer/svg.js:54]: Det er vel en group og ikke et array af groups

File: src/visualizer/svg.js [hunk 6]
```diff
@@ -43,38 +68,109 @@ const {
     /**
      * @summary Creates and appends a svg learningstyle to the svg element
      * @param {object} group the group array with students
-     * @param {number} learnStyleName the index of the learningstyle (used for linespaceing)
+     * @param {number} learnStyleName the name of the learningstyle
```
Comment (mads256h) [inline src/visualizer/svg.js:71]: Det er vel en string og ikke et number

File: src/visualizer/svg.js [hunk 7]
```diff
@@ -43,38 +68,109 @@ const {
...
+        for (const i in arrCircleSize) {
+            const xValue = circleXValue(group.students[i].criteria.learningStyles[learnStyleName], LSRANGEWIDTH, LSRANGEWIDTH / 2);
+            const studentColor = colorArr[i];
+            svg.appendChild(createCircle(xValue, yValue, studentColor, arrCircleSize[i]));
+        }
+    }
+
+    /**
+     * @summary Calls the createSubject function for each subject
+     * @param {object} group the group array with students
```
Comment (mads256h) [inline src/visualizer/svg.js:89]: group ikke group array

File: src/visualizer/svg.js [hunk 8]
```diff
@@ -43,38 +68,109 @@ const {
...
+            createSubject(group, subject.name, svg, yValue);
+            yValue += 2 * svgLineSpace;
+        }
+        return svg;
+    }
+
+    /**
+     * @summary Creates and appends a svg element with the subject to the svg element
+     * @param {object} group the group array with students
+     * @param {number} subjectName the name of the subject
```
Comment (mads256h) [inline src/visualizer/svg.js:106]: NaN

File: src/visualizer/svg.js [hunk 9]
```diff
@@ -43,38 +68,109 @@ const {
...
+        for (const i in arrCircleSize) {
+            const xValue = circleXValue(group.students[i].criteria.subjectPreference.subjects[subjectName].score, 1, 0);
+            const studentColor = colorArr[i];
+            svg.appendChild(createCircle(xValue, yValue, studentColor, arrCircleSize[i]));
+        }
+    }
+
+    /**
+     * @summary Calls the createSubject function for each subject
+     * @param {object} group the group array with students
```
Comment (mads256h) [inline src/visualizer/svg.js:124]: group not group array

**Review by mads256h — CHANGES_REQUESTED** [review]
små ting

File: src/visualizer/statisticsMath.js (hunk 2 above)
Comment (henneboy) [reply src/visualizer/statisticsMath.js:84]: @returns {object} ->@returns {number[]}

File: src/visualizer/statisticsMath.js (hunk 3 above)
Comment (henneboy) [reply src/visualizer/statisticsMath.js:93]: @returns {object} ->@returns {number[]}

File: src/visualizer/svg.js (hunk 4 above)
Comment (henneboy) [reply src/visualizer/svg.js:25]: * @param {object} group the group array with students
->
 * @param {object} group the group object with students

File: src/visualizer/svg.js (hunk 5 above)
Comment (henneboy) [reply src/visualizer/svg.js:54]: * @param {object} group the group array with students
->
     * @param {object} group the group object with students

File: src/visualizer/svg.js (hunk 6 above)
Comment (henneboy) [reply src/visualizer/svg.js:71]: Ja, fixed

File: src/visualizer/svg.js (hunk 7 above)
Comment (henneboy) [reply src/visualizer/svg.js:89]: Done

File: src/visualizer/svg.js (hunk 8 above)
Comment (henneboy) [reply src/visualizer/svg.js:106]: Now a String

File: src/visualizer/svg.js (hunk 9 above)
Comment (henneboy) [reply src/visualizer/svg.js:124]: group array -> group object

File: src/visualizer/statisticsMath.js (hunk 2 above)
Comment (mads256h) [reply src/visualizer/statisticsMath.js]: Fix teksten også "returns an array..."

File: src/visualizer/statisticsMath.js (hunk 3 above)
Comment (mads256h) [reply src/visualizer/statisticsMath.js]: Fix teksten også "returns an array..."

File: src/visualizer/svg.js [hunk 10]
```diff
@@ -15,66 +17,160 @@ const {
     const svgLineSpace = svgWidth * 0.04; // the amount of space between each line, by the width of the svg
     const svgLineWidth = svgWidth * 0.01;
     const svgTextSize = svgLineSpace * 0.5;
-    const RANGEWIDTH = 22;
+    const LSRANGEWIDTH = 22; // Learningstyle rangewith, from -11 to 11 = 22
+    const WAHRANGEWIDTH = 2; // Working At Home rangewith, from -1 to 1 = 2
```
Comment (mads256h) [inline src/visualizer/svg.js:21]: Inconsistent naming convention for const "global" variables

File: src/visualizer/svg.js [hunk 11]
```diff
@@ -188,10 +287,10 @@ const {
...
      * @param {number} a the first number
      * @param {number} b the second number
+     * @param {number} distance the maximum distance between the to elements
      * @returns {boolean} Returns boolean whether the distance between a & b is smaller than 'distance'
      */
-    function range(a, b){
-        const distance = 0.5;
-        return Math.abs(a - b) <= Math.abs(distance);
+    function range(a, b, distance){
+        return (Math.abs(a - b) <= Math.abs(distance));
```
Comment (mads256h) [inline src/visualizer/svg.js:294]: fjern paranteserne rundt om det hele

**Review by mads256h — CHANGES_REQUESTED** [review]

File: src/visualizer/svg.js (hunk 10 above)
Comment (henneboy) [reply src/visualizer/svg.js:21]: const svgTextSize = svgLineSpace * 0.5;
    const learningStyleRangeWidth = 22; // Learningstyle rangewith, from -11 to 11 = 22
    const workingAtHhomeRangeWidth = 2; // Working At Home rangewith, from -1 to 1 = 2

File: src/visualizer/svg.js (hunk 11 above)
Comment (henneboy) [reply src/visualizer/svg.js:294]: return Math.abs(a - b) <= Math.abs(distance);

File: src/visualizer/statisticsMath.js (hunk 2 above)
Comment (henneboy) [reply src/visualizer/statisticsMath.js]: Done, men du overså:
    /**
     * @summary Creates and returns a new array with the learningstyle values(-11 to 11)
     * of the learningstyle with the learningstyle(e.g. active, visual, sensing..) name provided in the learnStyleName argument
     * @param {object} group the group from which to get the data
     * @param {string} learnStyleName the name of the learningstyle e.g. "activeReflective"
     * @returns {number[]} returns an array with the students of the groups values in the given learningstyle, the key is the students idx in the groups.students object
     */
    function getLSValuesOfGroup(group, learnStyleName){
        return group.students.map((s) => s.criteria.learningStyles[learnStyleName]);
    }

File: src/visualizer/statisticsMath.js (hunk 3 above)
Comment (henneboy) [reply src/visualizer/statisticsMath.js]: New:
    /**
     * @summary Creates and returns a new array with the subject scores
     * of the subject with the subject name provided in the subjectName argument
     * @param {object} group the group from which to get the data
     * @param {string} subjectName the name of the subject
     * @returns {number[]} returns an array with the students of the groups values in the given subjects, the key is the students idx in the groups.students object
     */
    function getSubjectValuesOfGroup(group, subjectName){
        return group.students.map((s) => s.criteria.subjectPreference.subjects[subjectName].score);
    }

    /**
     * @summary Creates and returns a new array with the workingAtHome scores of the students in the group
     * @param {object} group the group from which to get the data
     * @returns {number[]} returns an array with the students of the groups values in workingAtHome, the key is the students idx in the groups
     */
    function getWorkingAtHomeOfGroup(group){
        return group.students.map((s) => s.criteria.workingAtHome);
    }

**Review by mads256h — APPROVED** [review]
lgtm

**Pull request merged**